frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/dialogs/GitPushDialog.java

```diff
@@ -64,6 +80,30 @@ private void push() {
     push(isForcePushRequired);
   }
 
+  /**
+   * Overridden to provide a way to perform an action once the push is completed successfully.
+   * Compare to the {@link com.intellij.dvcs.push.ui.VcsPushDialog#push(boolean)}
+   * which doesn't implement {@code onSuccess} callback.
+   */
+  @Override
+  @UIEffect
+  public void push(boolean forcePush) {
+
+    String title = DvcsBundle.message("push.process.pushing");
+    executeAfterRunningPrePushHandlers(new Task.Backgroundable(myProject, title, /* canBeCancelled */ true) {
+      @Override
+      public void run(ProgressIndicator indicator) {
+        myController.push(forcePush);
+      }
+
+      @Override
+      @UIEffect
+      public void onSuccess() {
+        doInUIThreadWhenReady.run();
```

Comment:
btw I think we should rather rename it from `doInUIThreadWhenReady` to `doInUiThreadWhenReady` someday everywhere

Reply:
lol, ok